backend/app/main.py

```diff
@@ -643,6 +648,106 @@ async def get_user(userid: str, current_user: dict = Depends(get_current_user)):
         raise HTTPException(
             status_code=500, detail=f"Internal Server Error. Please try again later. Error: {str(e)}")
 
+@app.get(
+    '/reviews',
+    response_model=ReviewGetResponse,
+    responses={
+        '400': {'model': ErrorResponse},
+        '404': {'model': ErrorResponse},
+        '500': {'model': ErrorResponse},
+    },
+    tags=['reviews'],
+)
+
+async def get_reviews(seller_id: str) -> Union[ReviewGetResponse, ErrorResponse]:
+    """
+    Get reviews for a seller
+    """
+    try:
+        seller = await users_collection.find_one({"_id": ObjectId(seller_id)})
+        if not seller:
+            raise HTTPException(status_code=404, detail="Seller not found")
+    
+        reviews = seller.get("reviews", [])
+        total_reviews = len(reviews)
+        average_rating = round(sum(r['rating'] for r in reviews)/total_reviews, 2) if total_reviews > 0 else 0.0
+
+        return ReviewGetResponse(
+            seller=seller_id,
+            total_reviews=total_reviews,
+            average_rating=average_rating,
+            reviews=reviews
+        )
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Internal server Error. {str(e)}")
+
+
+@app.post(
+    '/reviews',
+    response_model=None,
+    responses={
+        '201': {'model': ReviewPostResponse},
+        '400': {'model': ErrorResponse},
+        '404': {'model': ErrorResponse},
+        '500': {'model': ErrorResponse},
+    },
+    tags=['reviews'],
+)
+async def post_reviews(
+    body: ReviewPostRequest,
+) -> Optional[Union[ReviewPostResponse, ErrorResponse]]:
+    """
+    Create a review for a seller
+    """
+    try:
+        seller_id = body.seller_id
+
+        seller = await users_collection.find_one({"_id": ObjectId(body.seller_id)})
+        if not seller:
+            raise HTTPException(status_code=400, detail="Seller not found")
+        
+        #check if the reviewer has already left a review
+        reviews = seller.get("reviews")
+        if reviews:
+            for review in reviews:
+                if body.reviewer_id == review.get("reviewer_id"):
+                    raise HTTPException(status_code=404, detail="You have already left the user a review")
+
+
+        review = {
+            "reviewer_id": body.reviewer_id,
+            "rating": body.rating,
+            "comment": body.comment,
+            "timestamp": datetime.utcnow().isoformat(),
+        }
+        
+        await users_collection.update_one(
+            {"_id": ObjectId(seller_id)},
+            {"$push": {"reviews": review},
+             "$inc": {
+                 "rating_count": 1,
+                 "rating": body.rating
+             }}
+        )
+
+        #recalculate average
+        updated_seller = await users_collection.find_one({"_id": ObjectId(body.seller_id)})
```

Comment:
The idea here is recalculate the average rating using the formula outlined in Jira so we don't have to recompute based on all reviews on each request.

In other words:
- add a new review in the `reviews` collection
- each time a new review is added, recompute the average using: `(current rating * current rating count + new rating) / (current rating + 1)`